client/txn.go

```diff
@@ -289,6 +279,25 @@ func (txn *Txn) Run(b *Batch) error {
 	return sendAndFill(txn.send, b)
 }
 
+func (txn *Txn) commit() error {
+	return txn.sendEndTxnCall(true /* commit */)
+}
+
+func (txn *Txn) cleanup(err error) {
+	if err != nil {
+		if replyErr := txn.Rollback(); replyErr != nil {
+			log.Errorf("failure aborting transaction: %s; abort caused by: %s", replyErr, err)
+		}
+	}
+}
+
+// CommitNoCleanup is the same as Commit but will not attempt to clean
+// up on failure. It is exposed only for use in txn_correctness_test.go
+// because those tests manipulate transaction state at a low level.
+func (txn *Txn) CommitNoCleanup() error {
```

Comment:
I think that when we fix the way `sql` uses `Txn`, `Commit()` can go back to being `CommitNoCleanup`, and then all problems are solved. Besides, the client code here got more and more complicated and I'm happy @tamird put in the work to make it simple again. I'm also against tossing code around for no reason, but in this case I'm very much in favor of it.